plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -415,6 +422,211 @@ createWidget("discourse-poll-info", {
   }
 });
 
+function fetchGroupedResults(data) {
+  return ajax("/polls/grouped_poll_results.json", { data }).catch(error => {
+    if (error) {
+      popupAjaxError(error);
+    } else {
+      bootbox.alert(I18n.t("poll.error_while_fetching_voters"));
+    }
+  });
+}
+
+function transformUserFieldToLabel(fieldName) {
+  let transformed = fieldName.split("_");
+  transformed[0] =
+    transformed[0].charAt(0).toUpperCase() + transformed[0].slice(1);
+  return transformed.join(" ");
+}
+
+createWidget("discourse-poll-grouped-pies", {
+  tagName: "div.poll-grouped-pies",
+  buildAttributes(attrs) {
+    return {
+      id: `poll-results-grouped-pie-charts-${attrs.id}`
+    };
+  },
+
+  html(attrs) {
+    const fields = Object.assign({}, attrs.groupableUserFields);
+    later(() => {
```

Comment:
We will have to be careful to test this for memory leaks and to make sure that if the poll is re-rendered it still works.

Reply:
Yes I don't like this later all, we got to find a better way